client/client.go

```diff
@@ -1133,6 +1136,16 @@ type (
 		//  - serviceerror.NotFound
 		DescribeWorkflowExecution(ctx context.Context, workflowID, runID string) (*workflowservice.DescribeWorkflowExecutionResponse, error)
```

Comment:
Do we want to discourage/deprecate this in favor of either using `DescribeWorkflow` or `WorkflowService().DescribeWorkflowExecution`?

Reply:
I wasn't going to as part of this PR, I think if we want to deprecate these calls that just directly call the  `WorkflowService()` we should deprecate them all.

Reply:
:+1: Would support deprecating any high-level ones that don't really provide value over direct gRPC call, but definitely for this one we should deprecate the other one (can be separate PR)